src/components/trade/PriceImpact/utils.ts

```diff
@@ -0,0 +1,117 @@
+import BigNumber from 'bignumber.js'
+
+import { parseBigNumber } from 'utils'
+import { ONE_HUNDRED_BIG_NUMBER, ONE_BIG_NUMBER, ZERO_BIG_NUMBER } from 'const'
+
+import { PriceImpactArgs, PriceImpactArgsBase } from '../PriceImpact'
+
+// Limit as to where price colour changes
+const HIGH_PRICE_IMPACT_THRESHOLD = new BigNumber(5) // 5%
+const MID_PRICE_IMPACT_THRESHOLD = new BigNumber(3) // 3%
+const LOW_PRICE_IMPACT_THRESHOLD = ONE_BIG_NUMBER // 1%
+// TODO: remove for slippage toggle
+// 0.5% placeholder slippage
+const HIGH_PLACEHOLDER_SLIPPAGE = new BigNumber(1.05) // 5%
+const MID_PLACEHOLDER_SLIPPAGE = new BigNumber(1.03) // 3%
+
+enum ImpactLevel {
+  HIGH = 3,
+  MID = 2,
+  LOW = 1,
+  NONE = 0,
+}
+
+const determineImpactLevel = (impact: BigNumber | null): ImpactLevel => {
+  const impactLevel = ImpactLevel.NONE
+
+  switch (true) {
+    case impact?.gte(HIGH_PRICE_IMPACT_THRESHOLD):
+      return ImpactLevel.HIGH
+    case impact?.gte(MID_PRICE_IMPACT_THRESHOLD):
+      return ImpactLevel.MID
+    case impact?.lt(LOW_PRICE_IMPACT_THRESHOLD):
+      return ImpactLevel.LOW
+    default:
+      return impactLevel
+  }
+}
+
+// sets colour of strong tag presenting price impact
+const getImpactColourClass = (impact: BigNumber | null): string => {
+  const impactLevel = determineImpactLevel(impact)
+
+  switch (impactLevel) {
+    case ImpactLevel.HIGH:
+      return 'highImpact'
+    case ImpactLevel.MID:
+      return 'midImpact'
+    case ImpactLevel.LOW:
+      return 'lowImpact'
+    default:
```

Comment:
I prefer in this case no DEFAULT, and be exhaustive with all enum values

typescript is smart enough to know u forgot about one if you don't add the DEFAULT, however in your code, you won't be notified if you forgot about one enum value.

Reply:
done in #1550